cordova-lib/src/cordova/plugin.js

```diff
@@ -157,6 +157,7 @@ module.exports = function plugin(command, targets, opts) {
 
                             if (missingVariables.length) {
                                 shell.rm('-rf', pluginInfo.dir);
+                                events.emit('verbose', 'Removing ' + pluginInfo.dir + ' due to installation failure');
                                 var msg = 'Variable(s) missing (use: --variable ' + missingVariables.join('=value --variable ') + '=value).';
```

Comment:
What I'm suggesting is to remove the change at L160 and replacing it by ```events.emit(msg)``` because:
- The plugin isn't getting removed since it hasn't been installed in the first place
- It lets the user know the reason why the plugin won't be installed